package com.example.dexdogs.data.mapper

import com.example.dexdogs.data.model.DogDTO
import com.example.dexdogs.domain.model.Dog

class DogDTOMapper {

    private fun fromDogDTOToDogDomain(dogDTO: DogDTO): Dog {

        return Dog(
            dogDTO.id, dogDTO.index, dogDTO.name,
            dogDTO.type, dogDTO.heightFemale, dogDTO.heightMale,
            dogDTO.imageUrl, dogDTO.lifeExpectancy, dogDTO.temperament,
            dogDTO.weightFemale, dogDTO.weightMale
        )
    }

    fun fromDogDTOListToDogDomainList(dogDTOList: List<DogDTO>): List<Dog> {
        return dogDTOList.map { fromDogDTOToDogDomain(it) }
    }
}